RiotSwiftUI/Modules/Room/PollHistory/View/SegmentedPicker.swift

```diff
@@ -16,32 +16,33 @@
 
 import SwiftUI
 
-struct SegmentedPicker<Tag: Hashable>: View {
-    private let segments: [(String, Tag)]
-    private let selection: Binding<Tag>
+struct SegmentedPicker<Segment: Hashable & CustomStringConvertible>: View {
+    private let segments: [Segment]
+    private let selection: Binding<Segment>
     private let interSegmentSpacing: CGFloat
     
     @Environment(\.theme) private var theme
     
-    init(segments: [(String, Tag)], selection: Binding<Tag>, interSegmentSpacing: CGFloat) {
+    init(segments: [Segment], selection: Binding<Segment>, interSegmentSpacing: CGFloat) {
         self.segments = segments
         self.selection = selection
         self.interSegmentSpacing = interSegmentSpacing
     }
     
     var body: some View {
         HStack(spacing: interSegmentSpacing) {
-            ForEach(segments, id: \.1) { text, tag in
-                let isSelectedSegment = tag == selection.wrappedValue
+            ForEach(segments, id: \.hashValue) { segment in
+                let isSelectedSegment = segment == selection.wrappedValue
                 
                 Button {
-                    selection.wrappedValue = tag
+                    selection.wrappedValue = segment
                 } label: {
-                    Text(text)
+                    Text(segment.description)
                         .font(isSelectedSegment ? theme.fonts.headline : theme.fonts.body)
                         .underline(isSelectedSegment)
                 }
                 .accentColor(isSelectedSegment ? theme.colors.accent : theme.colors.primaryContent)
+                .accessibilityLabel(segment.description + (isSelectedSegment ? "-selected" : ""))
```

Comment:
I might be wrong, but maybe you can use `.accessibilityValue` here for selected?

Reply:
UI tests (like they are now) fail switching to `.accessibilityValue`.
Why do you think it would be a better option?

Reply:
Addressed here -> https://github.com/vector-im/element-ios/pull/7267/commits/25a52a24532cfa9bb6b27d663ceb1d80584ccf2b